ShopAppTests/App/Account/Controllers/ForgotPasswordViewControllerSpec.swift

```diff
@@ -0,0 +1,138 @@
+            viewController = UIStoryboard(name: StoryboardNames.account, bundle: nil).instantiateViewController(withIdentifier: ControllerIdentifiers.forgotPassword) as! ForgotPasswordViewController
+            
+            let repositoryMock = AuthentificationRepositoryMock()
+            let resetPasswordUseCaseMock = ResetPasswordUseCaseMock(repository: repositoryMock)
+            viewModelMock = ForgotPasswordViewModelMock(resetPasswordUseCase: resetPasswordUseCaseMock)
+            viewController.viewModel = viewModelMock
+            
+            titleLabel = self.findView(withAccessibilityLabel: "title", in: viewController.view) as! UILabel
+            descriptionLabel = self.findView(withAccessibilityLabel: "description", in: viewController.view) as! UILabel
+            emailTextFieldView = self.findView(withAccessibilityLabel: "email", in: viewController.view) as! InputTextFieldView
+            forgotPasswordButton = self.findView(withAccessibilityLabel: "forgotPassword", in: viewController.view) as! BlackButton
+            linkView = self.findView(withAccessibilityLabel: "link", in: viewController.view)
+        }
+        
+        describe("when view loaded") {
+            it("should have a correct view model type") {
+                expect(viewController.viewModel).to(beAKindOf(ForgotPasswordViewModel.self))
+            }
+            
+            it("should have a title with correct text") {
+                expect(viewController.title) == "ControllerTitle.ForgotPassword".localizable
+            }
+            
+            it("should have a close button") {
+                expect(viewController.navigationItem.rightBarButtonItem?.image) == #imageLiteral(resourceName: "cross")
+            }
+            
+            it("should have labels with correct texts") {
+                expect(titleLabel.text) == "Label.ForgotPassword.PasswordTitle".localizable
+                expect(descriptionLabel.text) == "Label.ForgotPassword.PasswordDescription".localizable
+            }
+            
+            it("should have text filed views with correct placeholders") {
+                expect(emailTextFieldView.placeholder) == "Placeholder.Email".localizable.required.uppercased()
+            }
+            
+            it("should have forgot password button with correct title") {
+                expect(forgotPasswordButton.title(for: .normal)) == "Button.Submit".localizable.uppercased()
+            }
+        }
+
+        describe("when email text changed") {
+            var disposeBag: DisposeBag!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+            }
+            
+            context("if it have at least one symbol in text field view") {
+                it("needs to enable forgot password button") {
+                    viewModelMock.makeForgotPasswordButtonEnabled()
+                    
+                    viewController.viewModel.resetPasswordButtonEnabled
```

Comment:
Can we avoid subscription in tests? Looks like we already have subscription between view controller and view model and just need to verify that button enabled status changed.